packages/definitions-parser/src/lib/module-info.ts

```diff
@@ -130,149 +130,141 @@ export function allReferencedFiles(
   entryFilenames: readonly string[],
   fs: FS,
   packageName: string,
-  baseDirectory: string
+  moduleResolutionHost: ts.ModuleResolutionHost,
+  compilerOptions: ts.CompilerOptions,
 ): { types: Map<string, ts.SourceFile>; tests: Map<string, ts.SourceFile>; hasNonRelativeImports: boolean } {
   const seenReferences = new Set<string>();
   const types = new Map<string, ts.SourceFile>();
   const tests = new Map<string, ts.SourceFile>();
+  // The directory where the tsconfig/index.d.ts is - i.e., may be a version within the package
+  const baseDirectory = path.resolve("/", fs.debugPath());
+  // The root of the package and all versions, i.e., the direct subdirectory of types/
+  const packageDirectory = baseDirectory.slice(0, baseDirectory.lastIndexOf(`types/${getMangledNameForScopedPackage(packageName)}`) + `types/${getMangledNameForScopedPackage(packageName)}`.length);
   let hasNonRelativeImports = false;
-  entryFilenames.forEach((text) => recur({ text, exact: true }));
+  entryFilenames.forEach((fileName) => recur(undefined, { text: fileName, kind: "path" }));
   return { types, tests, hasNonRelativeImports };
 
-  function recur({ text, exact }: Reference): void {
-    const resolvedFilename = exact ? text : resolveModule(text, fs);
-    if (seenReferences.has(resolvedFilename)) {
+  function recur(containingFileName: string | undefined, { text, kind, resolutionMode }: Reference): void {
+    // An absolute file name for use with TS resolution, e.g. '/DefinitelyTyped/types/foo/index.d.ts'
+    let resolvedFileName: string | undefined;
+    switch (kind) {
```

Comment:
Could this new patch of code still be in resolveModule? From a reviewer's perspective, it's easier to review, but from a future reader's perspective, having it here might be better.